rsqueakvm/storage_contexts.py

```diff
@@ -167,15 +167,25 @@ def is_privileged_index(self, n0):
             return (n0 == self.privileged_method_fields[0]) | (n0 == self.privileged_method_fields[1])
 
     def get_extra_data(self):
-        if self.extra_data is None:
-            self.extra_data = ExtraContextAttributes()
-        return self.extra_data
+        extra_data = self.extra_data_or_blockmethod
+        if extra_data is None or not isinstance(extra_data, ExtraContextAttributes):
```

Comment:
put the easier case first: `if isinstance(extra_data, ExtraContextAttributes): return...`